src/wallet/rpcwallet.cpp

```diff
@@ -3984,6 +3991,19 @@ UniValue z_shieldcoinbase(const UniValue& params, bool fHelp)
         max_tx_size = MAX_TX_SIZE_BEFORE_SAPLING;
     }
 
+    // If Sapling is not active, do not allow sending to a Sapling address.
+    if (!NetworkUpgradeActive(nextBlockHeight, Params().GetConsensus(), Consensus::UPGRADE_SAPLING)) {
+        auto res = DecodePaymentAddress(destaddress);
+        if (IsValidPaymentAddress(res)) {
```

Comment:
Should we decode the payment address and check its validity unconditionally?

Reply:
We do check it unconditionally, inside the constructor for `AsyncRPCOperation_shieldcoinbase`, which is run synchronously in this RPC method.